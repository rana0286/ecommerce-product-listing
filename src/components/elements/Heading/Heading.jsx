import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './Heading.scss';

class Heading extends PureComponent {
  render() {
    const { children, variation: Variation, ...rest } = this.props;

    return (
      <Variation {...rest} className={`csl-heading--${Variation}`}>
        {children}
      </Variation>
    );
  }
}

Heading.defaultProps = {
  variation: 'h1',
};

Heading.propTypes = {
  /** Sets the children */
  children: PropTypes.node.isRequired,
  /** Sets the variation */
  variation: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']),
};

export default Heading;
