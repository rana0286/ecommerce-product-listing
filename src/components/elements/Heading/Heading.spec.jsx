import React from 'react';
import { shallow } from 'enzyme';
import Heading from './Heading';
import renderer from 'react-test-renderer';

describe('src/components/elements/Heading', () => {
  it('Heading component should render', () => {
    const component = shallow(<Heading Variation="h1">Test heading</Heading>);
    const tree = renderer.
      create(<Heading Variation="h2">Test heading</Heading>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.find('h1').length).toBe(1);
  });
});
