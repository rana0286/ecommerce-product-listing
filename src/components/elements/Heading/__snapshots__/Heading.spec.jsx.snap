// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Header Header component should render 1`] = `
<h1
  Variation="h2"
  className="heading heading--h1"
>
  Test heading
</h1>
`;
